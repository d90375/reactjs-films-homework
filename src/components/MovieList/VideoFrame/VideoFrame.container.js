import React from "react";
import { useDispatch, useSelector } from "react-redux";
import VideoFrame from "./VideoFrame";
import { getTrailerDataSelector, isShowTrailerSelector } from "../../../modules/trailerData/trailerSelectors";
import { removeVideoFrame } from "../../../modules/trailerData/trailerActions";

const VideoFrameContainer = () => {
  const dispatch = useDispatch();
  const isShowVideoFrame = useSelector(isShowTrailerSelector);
  const { results } = useSelector(getTrailerDataSelector);

  const handleRemoveVideoFrame = () => {
    dispatch(removeVideoFrame());
  };

  return <>{isShowVideoFrame ? <VideoFrame trailer={results} onRemoveVideoFrame={handleRemoveVideoFrame} /> : null}</>;
};

export default VideoFrameContainer;
