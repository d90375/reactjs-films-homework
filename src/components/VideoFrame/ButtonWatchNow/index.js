import ButtonWatchNow from "./ButtonWatchNow.container";

export default ButtonWatchNow;
