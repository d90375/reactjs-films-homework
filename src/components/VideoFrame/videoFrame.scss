.videoFrame {
  align-items: flex-start;
  background-color: rgba(0, 0, 0, 0.9);
  display: flex;
  height: 100vh;
  justify-content: center;
  left: 0;
  padding-top: 5rem;
  position: fixed;
  top: 0;
  width: 100vw;
  z-index: 999;
}

.videoFrame__iframe {
  height: 35vw;
  width: 65vw;
}

.videoFrame__btn {
  background-color: transparent;
  border: none;
  color: tomato;
  cursor: pointer;
  font-size: 2rem;
  outline: none;
  padding: 1rem;

  &:hover {
    color: darken(tomato, 20%);
  }
}
