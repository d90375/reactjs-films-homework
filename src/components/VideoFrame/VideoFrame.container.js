import React from "react";
import VideoFrame from "./VideoFrame";
import { useTrailerActions, useTrailer, useTrailerData } from "../../modules/trailerData";

import Preloader from "../MovieList/Preloader";

const VideoFrameContainer = React.memo(() => {
  const { handleRemoveVideoFrame } = useTrailerActions();

  const { isLoadingTrailer, isFulfilledTrailer, hasErrorTrailer, errorTrailer } = useTrailer();
  const trailer = useTrailerData();

  return (
    <>
      {isLoadingTrailer && (
        <div className="videoFrame__wrap">
          <Preloader />
        </div>
      )}
      {isFulfilledTrailer && (
        <VideoFrame trailerKey={trailer?.results?.[0]?.key} onRemoveVideoFrame={handleRemoveVideoFrame} />
      )}
      {hasErrorTrailer && (
        <div className="error error__trailer">
          <span>Error {errorTrailer}</span>
        </div>
      )}
    </>
  );
});

export default VideoFrameContainer;

VideoFrameContainer.displayName = "VideoFrameContainer";
