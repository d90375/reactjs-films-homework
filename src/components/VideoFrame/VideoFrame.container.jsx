import React, { useCallback } from "react";
import { useDispatch, useSelector } from "react-redux";
import VideoFrame from "./VideoFrame";
import { getTrailerDataSelector, getTrailerSelector, removeVideoFrame } from "../../modules/trailerData";
import Preloader from "../MovieList/Preloader";

import styles from "./videoFrame.scss";

const VideoFrameContainer = React.memo(() => {
  const dispatch = useDispatch();

  const { isLoadingTrailer, isFulfilledTrailer, error } = useSelector(getTrailerSelector);
  const trailer = useSelector(getTrailerDataSelector);

  const handleRemoveVideoFrame = useCallback(() => dispatch(removeVideoFrame()), [dispatch]);

  return (
    <>
      {isLoadingTrailer && (
        <div className={styles.videoFrame}>
          <Preloader />
        </div>
      )}
      {isFulfilledTrailer && (
        <VideoFrame trailerKey={trailer?.results?.[0]?.key} onRemoveVideoFrame={handleRemoveVideoFrame} />
      )}
      {error && (
        <div className={styles.error}>
          <span>{`Error: ${error.message}`}</span>
        </div>
      )}
    </>
  );
});

export default VideoFrameContainer;

VideoFrameContainer.displayName = "VideoFrameContainer";
