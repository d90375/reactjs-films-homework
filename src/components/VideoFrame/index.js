import VideoFrame from "./VideoFrame.container";

export default VideoFrame;
