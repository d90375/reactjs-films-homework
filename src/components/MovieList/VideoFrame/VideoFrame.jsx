import PropTypes from "prop-types";
import React from "react";
import "./videoFrame.scss";

const VideoFrame = ({ trailer, onRemoveVideoFrame }) => {
  const link = trailer[0].key;
  return (
    <div className="videoFrame">
      <iframe
        className="videoFrame__iframe"
        title="trailer"
        src={`https://www.youtube.com/embed/${link}`}
        allowFullScreen
        frameBorder="0"
      />
      <button className="videoFrame__btn" type="button" onClick={onRemoveVideoFrame}>
        &#x2718;
      </button>
    </div>
  );
};

export default VideoFrame;

VideoFrame.propTypes = {
  onRemoveVideoFrame: PropTypes.func.isRequired,
  trailer: PropTypes.arrayOf(PropTypes.object)
};

VideoFrame.defaultProps = {
  trailer: [{}]
};
