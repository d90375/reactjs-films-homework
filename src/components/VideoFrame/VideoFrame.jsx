import PropTypes from "prop-types";
import React from "react";
import "./videoFrame.scss";

const VideoFrame = ({ trailerKey, onRemoveVideoFrame }) => {
  return (
    <div className="videoFrame">
      <iframe
        className="videoFrame__iframe"
        title="trailer"
        src={`https://www.youtube.com/embed/${trailerKey}`}
        allowFullScreen
        frameBorder="0"
      />
      <button className="videoFrame__btn" type="button" onClick={onRemoveVideoFrame}>
        &#x2718;
      </button>
    </div>
  );
};

export default VideoFrame;

VideoFrame.propTypes = {
  onRemoveVideoFrame: PropTypes.func.isRequired,
  trailerKey: PropTypes.string
};

VideoFrame.defaultProps = {
  trailerKey: ""
};
