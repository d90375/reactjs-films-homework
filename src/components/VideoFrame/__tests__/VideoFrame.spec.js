import React from "react";
import VideoFrame from "../VideoFrame";

const setUp = (props) => {
  return shallow(<VideoFrame {...props} />);
};

describe("VideoFrame component", () => {
  let component;

  beforeEach(() => {
    component = setUp();
  });

  it("should render VideoFrame component without props", () => {
    expect(component).toMatchSnapshot();
  });

  it("should render VideoFrame component with props", () => {
    component = setUp({ trailerKey: "key" });
    expect(component).toMatchSnapshot();
  });

  describe("should handler", () => {
    it("should call onHideView method", () => {
      const mockCallBack = jest.fn();
      component = setUp({ onRemoveVideoFrame: mockCallBack });
      expect(mockCallBack.mock.calls.length).toBe(0);
      component.find(".videoFrame__btn").simulate("click");
      expect(mockCallBack.mock.calls.length).toBe(1);
    });
  });
});
