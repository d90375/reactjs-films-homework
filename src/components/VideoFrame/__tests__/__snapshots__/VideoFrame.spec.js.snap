// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoFrame component should render VideoFrame component with props 1`] = `
<div
  className="videoFrame"
>
  <iframe
    allowFullScreen={true}
    className="videoFrame__iframe"
    frameBorder="0"
    src="https://www.youtube.com/embed/key"
    title="trailer"
  />
  <button
    className="videoFrame__btn"
    onClick={[Function]}
    type="button"
  >
    ✘
  </button>
</div>
`;

exports[`VideoFrame component should render VideoFrame component without props 1`] = `
<div
  className="videoFrame"
>
  <iframe
    allowFullScreen={true}
    className="videoFrame__iframe"
    frameBorder="0"
    src="https://www.youtube.com/embed/"
    title="trailer"
  />
  <button
    className="videoFrame__btn"
    onClick={[Function]}
    type="button"
  >
    ✘
  </button>
</div>
`;
